grammar/rainerscript.c

```diff
@@ -2122,6 +2122,50 @@ doFuncCall(struct cnffunc *__restrict__ const func, struct svar *__restrict__ co
 		if(bMustFree) free(str);
 		varFreeMembers(&r[1]);
 		break;
+	case CNFFUNC_FORMAT_TIME: {
+		time_t unixtime;
+		struct tm lt;
+		char   result[64];
+		char  *formatstr = NULL, *endptr = NULL;
+
+		cnfexprEval(func->expr[0], &r[0], usrptr);
+		cnfexprEval(func->expr[1], &r[1], usrptr);
+
+		str = (char*) var2CString(&r[0], &bMustFree);
+		formatstr = (char*) es_str2cstr(r[1].d.estr, NULL);
+
+
+		// TODO: Error checking, and account for different
+		//       possible sizes of time_t 32/64 bit.
+		errno = 0;
+		unixtime = (time_t) strtoll(str, &endptr, 10);
+
+		printf("STR: %s, FORMATSTR: %s, NUM: %ld\n", str, formatstr, unixtime);
+
+		localtime_r(&unixtime, &lt);
```

Comment:
Is it for sure a good idea to use localtime? For example, a RFC3164 timestamp does not contain the TZ information. So we can simply convert it as if it were UTC. When that value is used with localtime, the TZ offset is applied once again. Unfortunately there are many subtle issues with timezones...

Maybe it's OK to do this, and we only need to document the problem.

Reply:
Still considering the implications of using 'localtime'. Converting to syslogTimestamp and using the existing formatting methods could still end up being the way I go.

I admit strftime's use of TZ is annoying.

Reply:
IMHO the whole TZ system on Linux is unfortunately broken... How nice it would be just to be able to query the TZ database and tell the system what you want ;-) But complaining doesn't help and I am not yet up to propose a big clib patch ;-)